src/net/tls-impl.cc

```diff
@@ -537,7 +537,7 @@ class tls::reloadable_credentials_base {
                 }
             }
         }
-        future<fsnotifier::watch_token> add_watch(const sstring& filename, fsnotifier::flags flags = fsnotifier::flags::close_write|fsnotifier::flags::delete_self) {
+        future<fsnotifier::watch_token> add_watch(const sstring& filename, fsnotifier::flags flags = fsnotifier::flags::close_write|fsnotifier::flags::delete_self|fsnotifier::flags::move_to) {
```

Comment:
When executing
```
mv existing.crt.bak existing.crt
```
`existing.crt` should trigger a `delete_self`, which is already watched for, and the parent dir should issue a `move_to`.

I do not understand why adding `move_to` to the file watcher fixes the issue. What am I missing here?

The `add_dir_watch` function seems to be there to cater for moving files around, but for some reason it is only applied on rebuild and not on init.

Reply:
Testing this naively at the command line, it doesn't seem like this type of `mv` generates a delete event, or at least not guaranteed right away. 

which makes sense I think because inotify monitors the state of the inodes pointed to by the directory entry, rather than the `name->inode` mapping itself?

Reply:
Ah, never mind, this is a file watch. Now I'm confused too!

Reply:
Apologies, you are right. i should have been more careful in the description.
I was referring to the case where `existing.crt` already exists and we are trying to overwrite it with a move.

Reply:
well, if you're watching a directory for `delete` events, that type of move does _not_ immediately notify on my machine. but if you're watching a _file_ for `delete_self`, it does